### PR TITLE
Add CLI option to delete backup location

### PR DESCRIPTION
**Pull request by davidboxboro**
Signed-off-by: David Lu <davidboxboro@gmail.com>

Addresses issue #2168

### CODE REVIEW

**Comment by carlisia** [comment]
Thanks for this, pretty awesome!

I tried it locally and it's not working:
image


I can dig and see if it's my environment but wanted to ask, did this work for you?

**Comment by carlisia** [comment]
Also, instead of using the client-go rest api calls, maybe try changing to something equivalent to this: https://github.com/vmware-tanzu/velero/blob/main/pkg/cmd/cli/backuplocation/create.go#L175

That one uses the `sigs.k8s.io/controller-runtime/pkg/client` client, which we are moving all the api calls to.

**Comment by nrb** [comment]
@davidboxboro This is awesome, thanks!

Can you look into the CI failures and making the client changes Carlisia suggested?

File: pkg/cmd/cli/backuplocation/delete.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                        backupLocations = append(backupLocations, &res.Items[i])
+                }
+        }
+        if len(backupLocations) == 0 {
+                fmt.Println("No backup-locations found")
+                return nil
+        }
+
+        for _, s := range backupLocations {
+                err := o.Client.VeleroV1().BackupStorageLocations(s.Namespace).Delete(context.TODO(), s.Name, metav1.DeleteOptions{})
```
Comment (ashish-amarnath) [inline pkg/cmd/cli/backuplocation/delete.go:112]: We should also delete all the `backup` objects from the backup storage location before deleting the backup storage location.

**Comment by carlisia** [comment]
@davidboxboro if you can still work on this we'll reopen this PR, thank you.

**Pull request closed without merging**